code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -341,7 +341,7 @@
 			var/obj/O = new structurepath(target)
 			animate_flock_convert_complete(O)
 			playsound(target, "sound/misc/flockmind/flockdrone_build_complete.ogg", 40, 1)
-
+			O.AddComponent(/datum/component/flock_interest, F?.flock)
```

Comment:
Would it be safer instead just not adding the component for no Flock, or doesn't really matter?

Reply:
Safer in what way? Only drones can build barricades, and only in this method. If the drone has an associated flock, then it will be aligned with it, otherwise it's just a generic flock thing.

Reply:
Currently for all other cases of the component, the flock passed in should exist I think (called only by /datum/flock atm with passed in flock being src). The component doesn't have a null check on flock atm, could maybe be possible that the barricade action finishes at the same time that flock becomes null? Idk